### DISCUSSION TITLE
How to add X-FORWARDED-FOR for HTTP request?

### DISCUSSION THREAD

**Discussion by vanmanda** [ickerwx/tcpproxy#46] [Q&A]
If I understood the documentation, it should be possible to add the X-FORWARDED-FOR http-header using the http_post module. 
Can you give me an example on how to implement this?

Thank you

**Comment by ickerwx** [comment]
Hi,

no, that probably will not work. You would have to write something to parse the data stream and recognize HTTP messages. You might try and get it done somewhat similar to what I did in removegzip.py, try some very simple processing of the data and assume that a double \r\n indicates an HTTP message. This is not reliable though.

The proxy does not understand HTTP, it works on TCP payloads and does not do any higher layer decoding. Maybe try one of the HTTP proxies on github, I believe they might be better suited for your use case.
